main/networking/Runtime.ts

```diff
@@ -294,18 +308,25 @@ class UDPConn {
    * Sends messages when Gamepad information changes
    * or when 100 ms has passed (with 50 ms cooldown)
    */
-  sendGamepadMessages = (_event: IpcMainEvent, data: protos.GpState[]) => {
+  sendGamepadMessages = (_event: IpcMainEvent, data: protos.Input[]) => {
     if (data.length === 0) {
       data.push(
-        protos.GpState.create({
+        protos.Input.create({
           connected: false,
         })
       );
     }
 
-    const message = protos.GpState.encode(data[0]).finish();
-    this.logger.debug(`Dawn sent UDP to ${runtimeIP}`);
-    this.socket.send(message, UDP_SEND_PORT, runtimeIP);
+    const message = protos.UserInputs.encode({inputs: data}).finish();
+    let port = UDP_SEND_PORT;
+    let ip = runtimeIP;
+    if (runtimeIP.includes(':')) {
+      const split = runtimeIP.split(':');
+      ip = split[0];
+      port = Number(split[1]); 
+    }
```

Comment:
Same comment above about the ngrok stuff.

Reply:
Added a comment